File: benchmark_phase_c/bench_property.py
import sys


class Circle:
    def __init__(self, r):
        self.r = r

    @property
    def area(self):
        return 3.0 * self.r * self.r


def bench_property(n):
    c = Circle(2.0)
    total = 0.0
    i = 0
    while i < n:
        total = total + c.area
        i = i + 1
    return total


if __name__ == '__main__':
    n = int(sys.argv[1])
    print(bench_property(n))

File: benchmark_phase_c/bench_property.cs
using System;

class Circle
{
    public float r;
    public Circle(float r) { this.r = r; }
    public float area { get { return 3.0f * r * r; } }
}

class Program
{
    static float BenchProperty(int n)
    {
        var c = new Circle(2.0f);
        float total = 0.0f;
        int i = 0;
        while (i < n)
        {
            total = total + c.area;
            i = i + 1;
        }
        return total;
    }

    static void Main(string[] args)
    {
        int n = int.Parse(args[0]);
        Console.WriteLine(BenchProperty(n));
    }
}
